x/mongo/driver/topology/fsm.go

```diff
@@ -45,6 +49,8 @@ func (f *fsm) apply(s description.Server) (description.Topology, description.Ser
 		Kind:    f.Kind,
 		Servers: newServers,
 	}
+	f.compatible.Store(true)
```

Comment:
I think this ordering is problematic. If we're in this code because a server check finished for a server that was previously removed, the `findServer` call below will fail and we'll return. However, that means we won't go through the code that checks for compatibility. Now, the FSM thinks `compatible` is true, but it might be tracking a server with an unsupported version.

Reply:
Also, more generally, should we actually be using the `compatible` field somewhere? One thought is that we can check it in server selection to give back a meaningful error message. Right now, these compatibility errors are essentially ignored.

Reply:
Let's file an Improvement ticket to figure out how we want to use `compatibilityErr`. I still think it's important, but it's out of scope for this PR anyway.

Reply:
ack, filed GODRIVER-1659